js/game.js

```diff
@@ -0,0 +1,256 @@
+/*global window, console, document, Image
+*/
+
+// Game object constructor
+// "Revealing module pattern"
+function createGame () {
```

Comment:
surmised that the problems I faced with the Ammo and Ship objects when using the Functional Pattern for object creation were somehow related to the fact that there were multiple instances of these objects being created on the fly when the game was running.
For the Game and Utils objects, there is only a single instance running in each case (singleton?) and hence I didn't refactor to see how that would pan out. Will keep this in mind going forward.